Proofs/Bit_twiddling.lean

```diff
@@ -268,7 +268,7 @@ def parity32_impl (x : BitVec 32) : BitVec 32 :=
   (0x00006996#32 >>> x4) &&& 1#32
 
 theorem parity32_correct (x : BitVec 32) :
-  (parity32_spec x) = ((parity32_impl x).getLsb 0) := by
+  (parity32_spec x) = ((parity32_impl x).getLsbD 0) := by
```

Comment:
```suggestion
  (parity32_spec x) = ((parity32_impl x)[0]) := by
```
If the index is statically known to be in-bounds, using GetElem notation might allow for more rewrites

Reply:
Got the following counter-example when switching to using `getElem`:
```
The prover found a potential counterexample, consider the following assignment:
  x = 0xffffffff#32
  ofBool
    (27030#32 >>>
          ((x ^^^ x >>> 16 ^^^ (x ^^^ x >>> 16) >>> 8 ^^^ (x ^^^ x >>> 16 ^^^ (x ^^^ x >>> 16) >>> 8) >>> 4) &&&
            15#32) &&&
        1#32)[0] = 0x1#1
```
Is this unexpected?

Reply:
That is indeed unexpected. Let's stick with getLsbD, then.

Is the goalstate easily minimized? It'd be good to figure out why this happens regardless, so an MWE would be nice

Reply:
Yes. The following would fail
```
theorem crock (x : BitVec 32) :
  x.getLsbD 0 = x[0] := by bv_decide
```
with the error
```
The prover found a potential counterexample, consider the following assignment:
  x = 0xffffffff#32
  ofBool x[0] = 0x0#1
```